### PR TITLE
Add relative date calculations to date template variables

### LINKED ISSUE #41: Allow date manupulation in template variables
In my weekly notes, I would have template variables like

- {{date:YYYY-MM-DD}}
- {{date+1d:YYYY-MM-DD}}
- {{date+2d:YYYY-MM-DD}}
- ...

I'd use these variables in dataview queries to find files with text strings for "this week in history" or "this day in history" type features.

Would it be possible to add this, similar to what Periodic Notes has, as I'm keen to replace periodic notes with Journals?

### PR DESCRIPTION
**Pull request by arnoesterhuizen**
resolves #41

Feature inspired and based off of code from liamcain/obsidian-periodic-notes

### CODE REVIEW

**Comment by srg-kostyrko** [comment]
Thank you for your contribution.
Could you please also add information about this feature to `Variable reference` modal in `src/settings/ui/variable-reference.ts` and update readme

**Comment by arnoesterhuizen** [comment]
Thanks for pointing out where I missed an update. I've updated the code; anything else I can add to make this easy to use?

I'm a novice when it gets to plugin development, so thanks for helping me get some experience :)

File: README.md [hunk 1]
```diff
@@ -43,8 +43,8 @@ There variables can be used in note name template, note storage path, content of
 
 - `{{journal_name}}` - name of journal note belongs to
 - `{{note_name}}` - name of note
-- `{{start_date}}` - first day of week, month, quarter, year or interval depending on note type, formatted using date format from settings, format can be overridden using following syntax `{{start_date:format}}` where format is string using Moment.js format rules (like `{{start_date:YYYY-MM-DD}}`)
-- `{{end_date}}` - last day of week, month, quarter, year or interval depending on note type, formatting rules are the same as in `{{start_date}}`
+- `{{start_date}}` - first day of week, month, quarter, year or interval depending on note type, formatted using date format from settings, format can be overridden using following syntax `{{start_date:format}}` where format is string using Moment.js format rules (like `{{start_date:YYYY-MM-DD}}`). You can add date calculations, e.g. `{{start_date+5d:format}}` to add 5 days. You can add or subtract any number of days (`d`), weeks (`w`), months (`m`), or years (`y`).
```
Comment (srg-kostyrko) [inline README.md:46]: it is capital `M` for month (lower case `m` is for minutes, please update in other file as well

**Review by srg-kostyrko — CHANGES_REQUESTED** [review]
One small correction in descriptions and it will be good to go

**Comment by arnoesterhuizen** [comment]
I've pushed a tweak to the readme and settings UI, and rather than writing a whole paragraph in a small space, included links to the momentjs reference. Hopefully this should make it easy to understand without cluttering the interface.

**Review by srg-kostyrko — APPROVED** [review]

**Pull request merged**